hash.go

```diff
@@ -358,6 +359,24 @@ func (h *md5Marshal) UnmarshalBinary(b []byte) error {
 	return nil
 }
 
+func (h *md5Marshal) AppendBinary(buf []byte) ([]byte, error) {
+	binaryData, err := h.MarshalBinary()
```

Comment:
I appreciate that you intended to implement the new methods with the few modifications possible. On the other hand, implementing `AppendBinary` in terms of `MarshalBinary` doesn't achieve the sole benefit of the `encoding.BinaryAppender` interface: not allocating a new slice every time it is called.

It would be much better if the code inside `MarshalBinary` is moved to `AppendBinary` (with the necessary changes), and then `MarshalBinary` is updated to call `AppendBinary`.

Reply:
Yes, thanks for recommending this approach 👍. I applied it and now looks much better.